### ISSUE TITLE
Current min/max, daily list days are pulling past data?

### ISSUE THREAD

**Issue by 0-5-0** [Akylas/oss-weather#161]
Looking at the "Current" weather min/max temperature forecast at the top right in the app the numbers were off with today's (Saturday's) weather forecast shown in the "Daily" list below.

I selected "Show current day in daily list" under "Weather Data" preferences and it's listing "Friday" as the current day at the bottom, you can then see the "Current/Daily" min/max temperature at the top is being pulled from Friday (yesterday) instead of Saturday (today).

My phone's date is Saturday, the app shows Saturday in the top right but it's pulling data from Friday for that "Current" section and is listing Friday at the bottom even though the day is past.

I cleared the apps cache/data and the issue occurs again. I also tried one location in Canada and one in France in case it was a weird timezone thing but the issue occurs in both locations.

<img src="https://github.com/Akylas/oss-weather/assets/112411524/c1505d48-8667-4df8-a079-55e74915c89e" width=50% height=50%>

**Comment by farfromrefug** [comment]
@0-5-0 i am yet to reproduce this. But there is clearly an issue! does it happen all the time?  Is it good sometimes? (with "Show current day in daily list")

**Comment by 0-5-0** [comment]
I hadn't noticed it before really, it seems to be a consistent issue so far now that I've noticed it.

Currently it's showing Monday 10/15 for the temperature looking at Grenoble in the Current view and it still shows Sun 10/15 under the Daily list (even though we're Monday). Monday's min/max in the daily list is 10/12 for Grenoble currently which is what I would expect to see under the Current min/max view.

I had never used the "Show current day in daily list" function, it was always off but having it selected now it makes the problem more obvious when comparing min/max. I tried turning them off/on in different orders, current day, current day precipitation, feels like, but it's not making a difference.

**Comment by 0-5-0** [comment]
I installed the app on another phone as well, fresh install, added Grenoble as the location, the min/max for Monday in the top right doesn't match the min/max for Monday in the daily list. I go in and select "Show current day in day list" and it brings up "Sunday" in the daily list but Monday is still in the Current forecast and it's pulling min/max from Sunday.

**Comment by farfromrefug** [comment]
@0-5-0 OK thanks will try to reproduce that all. Did not manage yet.
You are using Grenoble because you know I always use it? 
Thanks

**Comment by 0-5-0** [comment]
Exactly, it was in your original screenshots for the main app page :), so I try to test with both in case it makes a difference

**Comment by farfromrefug** [comment]
> I installed the app on another phone as well, fresh install, added Grenoble as the location, the min/max for Monday in the top right doesn't match the min/max for Monday in the daily list. I go in and select "Show current day in day list" and it brings up "Sunday" in the daily list but Monday is still in the Current forecast and it's pulling min/max from Sunday.

Something is weird in what you are saying. On a fresh install , as "Show current day in day list" is off by default, you should not see monday in the daily list view at the bottom (my case here).
It almost feels like only the date are offset.

Your phone is english?

**Comment by 0-5-0** [comment]
Yes it is off by default, with everything on default just adding the location, I see Monday everywhere but the min/max temperature listed at the top in Monday "Current" doesn't match the min/max temperature listed in Monday "Daily list".

As a troubleshooting step I then go into preferences and enable "Show current day in day list" but the day that shows up as the first one in day list is the yesterday's date (Sunday). I can then see the Monday's min/max temperature is getting pulled from Sunday (yesterday's date) which is now shown under the Daily list (since I enabled Show current day in day list).

**Comment by 0-5-0** [comment]
If I change the language of the app to Francais or my phone OS language to Francais France the issue is still there it just says Dimanche when I enable "Show current day in day list" but the current day should be Monday in the day list.

**Comment by farfromrefug** [comment]
> Yes it is off by default, with everything on default just adding the location, I see Monday everywhere but the min/max temperature listed at the top in Monday "Current" doesn't match the min/max temperature listed in Monday "Daily list".
> 
> As a troubleshooting step I then go into preferences and enable "Show current day in day list" but the day that shows up as the first one in day list is yesterday's date (Sunday). I can then see the Monday's min/max temperature is getting pulled from Sunday (yesterday's date) which is now shown under the Daily list (since I enabled Show current day in day list).
> 
> My phone is English, yes.

Sorry to bother you with that. COuld you:
* clear storage data of OSS Weather
* start recording your screen
* start the app, add Grenoble ... 
That way i can see how you do it and try to reproduce it

**Comment by 0-5-0** [comment]
Cleared app settings/cache, launched it, added Grenoble, you can see Monday is in Current forecast, Monday is in Daily List but the min/max temperature for both doesn't match. If I go into settings and enable "Show current day in daily list" it shows Sunday as the current day.

https://github.com/Akylas/oss-weather/assets/112411524/00985334-34b7-42e4-8a04-e07c8be2761a

**Comment by farfromrefug** [comment]
Ok this is what i get and what i suspected. For some reason the date is wrong. What you are seeing as Mon is actually tuesday. It is all decaled and this is why you see sunday appear.
What s your timezone? I think the issue lies there
![Screenshot_1715024769](https://github.com/Akylas/oss-weather/assets/655344/77340c65-725f-45d5-9640-9c46496b65aa)

**Comment by 0-5-0** [comment]
My timezone is GMT-04:00 Canada/Toronto, I tried Paris/GMT+2:00, clearing the app from the system tray, relaunching it and it does show what your screenshot shows when I have "show current day in daily list", if I change it back to GMT-04:00, clear the app from system tray and re-launch then it goes back to showing "Sunday" as the current day with "show current day in daily list", it's 15:54/3:54pm Monday here so it's been Monday for a while, not sure why it's doing that :)

**Comment by farfromrefug** [comment]
@0-5-0 thanks it is a timezone issue. i have to wrap my head around this and i ll fix it tomorrow. THanks a lot for your help!

**Comment by farfromrefug** [comment]
@0-5-0 actually fixed  :P new version is building.
Now it raises another issue that have been asked before. When you see Grenoble from Toronto your are seeing Toronto times where we would expect to see Grenoble times.
Next on my list!

**Comment by 0-5-0** [comment]
It might also be related to the Open-Meteo issue you opened: https://github.com/open-meteo/open-meteo/issues/642

**Comment by 0-5-0** [comment]
Looks great now, min/max and the "Current" day in current forecast matches the "Current" day in daily list view as well as when I have "Show current day in day view" selected. Thanks for finding and fixing it so quick!

Side note, I tried the new line vs default view for weather data.. I can't decide which one I like more but it's nice to have the choice. I'm not sure if adding a small "n/a" (not applicable/not available) would be an option for the different weather data entries, even configurable in the settings similar to UV index isn't show below 1, for those who might want to see all icons at all times, so that it could still show the icon and that a value was "requested" but the model didn't return it for that day.

It would be possible to always show the different weather data in the same location based on your sorting that way which might make it easier to read. Wind would always be spot 1, AQI spot 2, Precipitation spot 3, and they wouldn't move around depending on if precipitation data is available for that specific day (if that all makes sense).

**Comment by 0-5-0** [comment]
I do like the line one better having looked at it all evening :), I'm thinking now having all the icons show up even with n/a might not be great since you have a lot of data points now but I'm still wondering if there might be a way to tell each data point to sort in a specific location (1, skip 2, 3, 4, skip 5), up to your max number of sections for the lines view. I'm not sure what would be best honestly.

I noticed for the single clicking weather data entries in settings vs manually sorting them that it doesn't seem to save the single clicks, when I go back to weather data settings the entries I single clicked go back to where they were and the ones I dragged manually stay where I dragged them, I haven't tried clearing my app settings though.
